tests/languages/liquid/nested_feature.test

```diff
@@ -0,0 +1,151 @@
+<script>
+  // The drop is highlighted because liquid will render inside the comment.
+  // As such, it isn't a "real" comment w.r.t. liquid.
+  // comment {{ product.value }}
+  function main(arg1) {
+    if (condition) {
+      {% if condition %}
+        const foo = {{ variable | filter: arg1 | json }}
+      {% endif %}
+    }
+  }
+</script>
+
+{% javascript %}
+  // The drop is highlighted because liquid will render inside the comment.
+  // As such, it isn't a "real" comment w.r.t. liquid.
+  // comment {{ product.value }}
+  function main(arg1) {
+    if (condition) {
+      {% if condition %}
+        const foo = {{ variable | filter: arg1 | json }}
+      {% endif %}
+    }
+  }
+{% endjavascript %}
+
+----------------------------------------------------
+
+[
+	["liquid", [
+		["script", [
+			["tag", "<script>"],
+
+			["javascript", [
+				["comment", "// The drop is highlighted because liquid will render inside the comment."],
+
+				["comment", "// As such, it isn't a \"real\" comment w.r.t. liquid."],
+
+				["comment", "// comment {{ product.value }}"],
```

Comment:
From the comments above, I guess that this is supposed to be a liquid section, right?

Reply:
Yeah, definitely still a work in progress! Is there a better way of reusing the existing delimiters and extending markup/markup-templating?

Reply:
Unfortunately, there isn't right now. I have a PR open adding general templating functions that would allow us to correctly handle these nested cases, but that PR will have to wait for v2 and ESM.

Seems like you have to make a tradeoff here. We can either tokenize JS correctly and miss some Liquid sections, or tokenize Liquid correctly and miss some JS. I would tend to tokenizing Liquid correctly since that's the main goal here. This is also easier to do since we just have to move the `liquid` token a little:

```js
		inside: {
			'tag': /.../,
			'liquid': {
				pattern: /\{%.*%\}|\{\{.*\}\}/,
				inside: Prism.languages.liquid,
			},
			'javascript': {
				alias: ['js', 'javascript', 'language-javascript'],
				pattern: /\S+(?:\s+\S+)*/,
				inside: Prism.languages.javascript
			},
		}
```